CLI tools/MultiMC to Git.py

```diff
@@ -7,26 +7,20 @@
 
 # Functions
 
-def copy_file(from_path, to_path, from_desc, to_desc):
+def copy_file(from_path: str, to_path: str, from_desc: str, to_desc: str) -> None:
     if os.path.isfile(from_path):
         shutil.copy2(from_path, to_path)
         print("Copied " + from_desc + " to " + to_desc)
     else:
         print("Skipped " + from_desc + " copying to " + to_desc + ", didn't exist")
 
-def copy_to_archive(from_path, to_path, archive_path):
+def copy_to_archive(from_path: str, to_path: str, archive_path: str) -> None:
 
-    files = []
-
-    with zipfile.ZipFile(archive_path) as archive:
-        for zipinfo in archive.infolist():
-            if zipinfo.filename != to_path:
-                files.append((zipinfo.filename, archive.read(zipinfo.filename)))
-
-    with zipfile.ZipFile(archive_path, "w") as archive:
-        for filename, content in files:
-            archive.writestr(filename, content)
-        archive.write(from_path, to_path)
+    with zipfile.ZipFile(archive_path, "r") as inp, zipfile.ZipFile(archive_path, "w") as out:
+        files = {zipinfo.filename: inp.read(zipinfo.filename) for zipinfo in inp.infolist() if zipinfo.filename != to_path}

```

Comment:
It's nasty. And even if it optimizes anything, it's not worth creating such an unreadable mess.

Reply:
I would rather suggest revert changes and shorten file reading
```python

if (fname := zipinfo.filename) != to_path:
    files.append((fname, archive.read(fname)))

Reply:
Well I pitched a dict comprehension which is cleaner than using KV tuples

Reply:
one beeeg line with multiple statements is cleaner than line by line loops?